app/controllers/group_assignment_invitations_controller.rb

```diff
@@ -11,16 +12,31 @@ class GroupAssignmentInvitationsController < ApplicationController
 
   before_action :authorize_group_access, only: [:accept_invitation]
 
-  before_action :ensure_github_repo_exists, only: [:successful_invitation]
+  before_action :ensure_github_repo_exists,    only: %i[setup setup_progress successful_invitation]
+  before_action :ensure_authorized_repo_setup, only: %i[setup setup_progress]
```

Comment:
We should do the authorized check before the ensure_github_repo_exists check. The github_repo_exists check has the possibility of causing side effects.